src/libs/actions/Session/index.js

```diff
@@ -204,55 +204,84 @@ function resendLinkWithValidateCode(login = credentials.login) {
     API.write('RequestNewValidateCode', {email: login}, {optimisticData, successData, failureData});
 }
 
+function generateResponseData() {
+    const errorMessage = Localize.translateLocal('loginForm.cannotGetAccountDetails');
+    return {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.ACCOUNT,
+                value: {
+                    ...CONST.DEFAULT_ACCOUNT_DATA,
+                    isLoading: true,
+                    message: null,
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.ACCOUNT,
+                value: {
+                    isLoading: false,
+                },
+            },
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.CREDENTIALS,
+                value: {
+                    validateCode: null,
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: ONYXKEYS.ACCOUNT,
+                value: {
+                    isLoading: false,
+                    // eslint-disable-next-line rulesdir/prefer-localization
+                    errors: {
+                        [DateUtils.getMicroseconds()]: errorMessage,
+                    },
+                },
+            },
+        ],
+    };
+}
+
 /**
  * Checks the API to see if an account exists for the given login
  *
  * @param {String} login
  */
+
 function beginSignIn(login) {
-    const optimisticData = [
-        {
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.ACCOUNT,
-            value: {
-                ...CONST.DEFAULT_ACCOUNT_DATA,
-                isLoading: true,
-                message: null,
-            },
-        },
-    ];
+    const {optimisticData, successData, failureData} = generateResponseData();
+    API.read('BeginSignIn', {email: login}, {optimisticData, successData, failureData});
+}
 
-    const successData = [
-        {
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.ACCOUNT,
-            value: {
-                isLoading: false,
-            },
-        },
-        {
-            onyxMethod: Onyx.METHOD.MERGE,
-            key: ONYXKEYS.CREDENTIALS,
-            value: {
-                validateCode: null,
-            },
-        },
-    ];
+function handleAppleAuthApiResponse(token) {
+    const {optimisticData, successData, failureData} = generateResponseData();
+    // eslint-disable-next-line rulesdir/no-api-side-effects-method
+    API.makeRequestWithSideEffects('AuthenticateApple', {token}, {optimisticData, successData, failureData})
```

Comment:
I noticed the backend for this command is expecting `idToken` instead of `token`:

```suggestion
    API.makeRequestWithSideEffects('AuthenticateApple', {idToken: token}, {optimisticData, successData, failureData})
```

Reply:
Fixed, ty!